File: backend/routers/v1/harbor.go
package v1

import (
	"devops-platform/internal/middleware"
	harborAPI "devops-platform/internal/modules/harbor/api"

	"github.com/gin-gonic/gin"
)

func registerHarbor(r *gin.RouterGroup) {
	g := r.Group("/harbor")
	queryPermission := middleware.RequirePermission("harbor", "list")
	updatePermission := middleware.RequirePermission("harbor", "update")

	// Config
	g.GET("/configs", queryPermission, harborAPI.ListHarborConfigs)
	g.POST("/configs", updatePermission,
		middleware.SetAuditOperation("Harbor 配置保存"),
		harborAPI.SaveHarborConfig)
	g.PUT("/configs/:id", updatePermission, harborAPI.SaveHarborConfig)
	g.DELETE("/configs/:id", updatePermission,
		middleware.SetAuditOperation("Harbor 配置删除"),
		harborAPI.DeleteHarborConfig)
	g.POST("/configs/test", queryPermission, harborAPI.TestHarborConnection)

	// Projects
	g.GET("/projects", queryPermission, harborAPI.ListProjects)

	// Repositories
	g.GET("/projects/:projectName/repos", queryPermission, harborAPI.ListRepositories)

	// Artifacts
	g.GET("/projects/:projectName/repos/:repoName/artifacts", queryPermission, harborAPI.ListArtifacts)
	g.DELETE("/projects/:projectName/repos/:repoName/artifacts", updatePermission,
		middleware.SetAuditOperation("删除 Harbor artifact"),
		harborAPI.DeleteArtifact)
}

File: frontend/src/api/harbor.js
import request from './request'

export const listHarborConfigs = (params) => request.get('/harbor/configs', { params })
export const saveHarborConfig = (data) => request.post('/harbor/configs', data)
export const updateHarborConfig = (id, data) => request.put(`/harbor/configs/${id}`, data)
export const deleteHarborConfig = (id) => request.delete(`/harbor/configs/${id}`)
export const testHarborConnection = (data) => request.post('/harbor/configs/test', data)

export const listProjects = (params) => request.get('/harbor/projects', { params })
export const listRepositories = (projectName, params) => request.get(`/harbor/projects/${encodeURIComponent(projectName)}/repos`, { params })
export const listArtifacts = (projectName, repoName, params) => request.get(`/harbor/projects/${encodeURIComponent(projectName)}/repos/${encodeURIComponent(repoName)}/artifacts`, { params })
export const deleteArtifact = (projectName, repoName, data) => request.delete(`/harbor/projects/${encodeURIComponent(projectName)}/repos/${encodeURIComponent(repoName)}/artifacts`, { data })
